pypesto/store/hdf5.py

```diff
@@ -29,6 +29,8 @@ def write_array(f: h5py.Group,
     elif all(isinstance(x, str) for x in values):
         write_string_array(f, path,
                            values)
+    else:
```

Comment:
Not sure, but this looks potentially problematic. @jvanhoefer @plakrisenko @MerktSimon What was the reason to explicitly use the "write_x_array" functions?